### PR TITLE
Adds support for go modules.

### PR DESCRIPTION
**Pull request by kpurdon**
This PR adds basic support fo [go modules](https://github.com/golang/go/wiki/Modules) by adding the required `go.mod` and `go.sum` files which were generated with the `go mod init && go mod tidy` commands.

Other Updates:
- TravisCI will opt-in to using modules using the `GO111MODULES` environment variable. (only affects Go 1.11.x)
- Removes Go 1.7 from TravisCI (go mysql no longer supports it)

### CODE REVIEW

**Comment by l3pp4rd** [comment]
Hi, this would force downloading these dependencies, just in order to run the tests. The library in general has no dependencies. I'll maybe consider moving the tests to a separate directory where it could be more like integration tests. and leaving go test to just run unit tests. I understand now that testing all modules would fail now with `go mod` so we need to make some changes, but this is not the right way in my opinion.

**Comment by kpurdon** [comment]
Thanks for the response!

TLDR: I do think this is the minimal/correct first step toward module-izing this repo, and have linked some info. However, it's your call, so feel free to close this and I'll stick with my module-enabled fork at [kpurdon/go-txdb](https://github.com/kpurdon/go-txdb).

> Hi, this would force downloading these dependencies, just in order to run the tests.

This is true (for now), but of note only for people explicitly using go mod (via `GO111MODULES=on`). At the current time the new `go get` w/ modules enabled is basically the same as `go get -t` w/o modules enabled.

There has been some conversation about this issue (test-only deps): https://github.com/golang/go/issues/26913#issuecomment-411976222.

> I'll maybe consider moving the tests to a separate directory where it could be more like integration tests. and leaving go test to just run unit tests.

I'm not quite sure that would solve the problem, but could be missing something. Unless it was a separate repo of go-txdb integration tests they would still be a part of the module.

>  I understand now that testing all modules would fail now with go mod so we need to make some changes, but this is not the right way in my opinion.

I'm not quite sure I understand this either, but maybe thats ok.
